src/helpers/config.ts

```diff
@@ -15,6 +16,17 @@ export interface AutoWaitConfig {
 
 export type Config = Partial<ReadConfig & SignerConfig>;
 
+export async function checkWait(
+  config: Config & Partial<AutoWaitConfig>,
+  receipt: WaitableTransactionReceipt
+): Promise<WaitableTransactionReceipt> {
+  if (config.autoWait ?? false) {
+    const waited = await receipt.wait();
+    return { ...receipt, ...waited };
+  }
+  return receipt;
```

Comment:
We had this exact block of code used in several places, so I setup a helper for it.